libs/dst/ProcPathStasCollection.py

```diff
@@ -0,0 +1,98 @@
+# -*- coding: utf-8 -*-
+# !/usr/bin/python
+#
+# Copyright (c) 2022 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as published
+# by the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+# GNU Affero General Public License for more details.
+# You should have received a copy of the GNU Affero General Public License
+# along with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+# For any questions about this software or licensing,
+# please email opensource@seagate.com or cortx-questions@seagate.com.
+#
+
+"""Procpath memory stats collector """
+
+import logging
+from multiprocessing import Process
+from commons.helpers.pods_helper import LogicalNode
+
+proc_cmd = "pid=$(echo $(pgrep m0d; pgrep radosgw; pgrep hax) | sed -z 's/ /,/g'); procpath " \
+           "record -i 45 -d s3server-$(date +%d%m_%H%M%S).sqlite -p $pid"
+
+pid_watch_list = ['m0d', 'radosgw', 'hax']
+
+# check and set pytest logging level as Globals.LOG_LEVEL
+LOGGER = logging.getLogger(__name__)
+
+
+class EnableProcPathStatsCollection:
+
+    def __init__(self, cmn_cfg):
+        """Initialize connection to Nodes or Pods."""
+        self.cmn_cfg = cmn_cfg
+        self.nodes = cmn_cfg["nodes"]
+        self.worker_node_list = list()
+        for node in self.nodes:
+            if node["node_type"].lower() == "worker":
+                node_obj = LogicalNode(hostname=node["hostname"],
+                                       username=node["username"],
+                                       password=node["password"])
+                self.worker_node_list.append(node_obj)
+        self.stat_collection = []
+
+    @staticmethod
+    def stat_collection_cmd(worker: LogicalNode, cmd: str):
+        """command for Procpath collection"""
+        LOGGER.debug("executing procpath cmd for stat collection")
+        worker.execute_cmd(cmd=cmd)
+
+    @staticmethod
+    def collect_pids(worker, watch_list):
+        """function to collect pids """
+        for proc in watch_list:
+            res = worker.execute_cmd("echo $(pgrep {})".format(proc))
+            logging.info(f'{proc} PIDs {res}')
+            with open("{}".format(worker.hostname), 'a') as fp:
+                fp.write(f'{proc} PIDs {res}')
+
+    def setup_requirement(self):
+        """Install required modules on worker nodes for procpath collection"""
+        LOGGER.debug("installing required modules")
+        for worker_node in self.worker_node_list:
+            worker_node.execute_cmd("pip install Procpath apsw-wheels")
```

Comment:
So we can directly install and no errors are expected? what if installation fails? how we handle it?